Push0 rematerializer default sequence (#14468)
ethereum/solidity: Changelog.md

Compiler Features:
 * Yul Optimizer: If ``PUSH0`` is supported, favor zero literals over storing zero values in variables.

>  * Yul Optimizer: Run ``Rematerializer`` and ``UnusedPruner`` steps as part of the default clean-up sequence.

Reviewer: ```suggestion
 * Yul Optimizer: Run the ``Rematerializer`` and ``UnusedPruner`` steps at the end of the default clean-up sequence.
```
